agent/engine/docker_task_engine_test.go

```diff
@@ -513,6 +512,87 @@ func TestCreateContainerForceSave(t *testing.T) {
 	}
 }
 
+// TestTaskTransitionWhenStopContainerTimesout tests that task transitions to stopped
+// only when terminal events are recieved from docker event stream when
+// StopContainer times out
+func TestTaskTransitionWhenStopContainerTimesout(t *testing.T) {
+	ctrl, client, taskEngine := mocks(t, &defaultConfig)
+	defer ctrl.Finish()
+	ttime.SetTime(dte_test_time)
+
+	sleepTask := testdata.LoadTask("sleep5")
+
+	eventStream := make(chan DockerContainerChangeEvent)
+
+	dockerEvent := func(status api.ContainerStatus) DockerContainerChangeEvent {
+		meta := DockerContainerMetadata{
+			DockerId: "containerId",
+		}
+		return DockerContainerChangeEvent{Status: status, DockerContainerMetadata: meta}
+	}
+
+	client.EXPECT().ContainerEvents(gomock.Any()).Return(eventStream, nil)
+	containerStopTimeoutError := DockerContainerMetadata{Error: &DockerTimeoutError{transition: "stop", duration: stopContainerTimeout}}
+	for _, container := range sleepTask.Containers {
+
+		client.EXPECT().PullImage(container.Image, nil).Return(DockerContainerMetadata{})
+
+		dockerConfig, err := sleepTask.DockerConfig(container)
+		if err != nil {
+			t.Fatal(err)
+		}
+		client.EXPECT().CreateContainer(dockerConfig, gomock.Any(), gomock.Any()).Do(func(x, y, z interface{}) {
+			go func() { eventStream <- dockerEvent(api.ContainerCreated) }()
+		}).Return(DockerContainerMetadata{DockerId: "containerId"})
+
+		// StartContainer returns timeout error. This should cause the engine
+		// to transition the task to STOPPED and to stop all containers of
+		// the task
+		client.EXPECT().StartContainer("containerId").Return(DockerContainerMetadata{Error: &DockerTimeoutError{}})
+
+		// StopContainer times out as well
+		client.EXPECT().StopContainer(gomock.Any()).Do(func(id string) {
```

Comment:
Can you put the two `StopContainer` expectations inside a `gomock.InOrder` to enforce ordering?